### DISCUSSION TITLE
nnn icon support

### DISCUSSION THREAD

**Discussion by daviehh** [Homebrew/discussions#4901] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.20-2-gd72ee37
ORIGIN: https://github.com/Homebrew/brew
HEAD: d72ee37f60e287e833ef9b1882bc6ca8ded01aea
Last commit: 10 hours ago
Core tap JSON: 13 Nov 21:58 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: subl
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.42.1 => /opt/homebrew/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1.1-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: 15.0.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

For the terminal file manager nnn, the current default flags may lead to icons shown as ? for typical setups:

Since the commit https://github.com/Homebrew/homebrew-core/commit/c65f99050610689dd00e70a837c12885813822ad, the compile arguments
```
args = %w[
       O_EMOJI=1
       O_NERD=1
       O_ICONS=1
     ]
```
are added. However, these flags are sort of mutually exclusive and thus the current nnn bottle shows icons as `?`.

Basicially, `nnn` shows icon with three different approaches [doc](https://github.com/jarun/nnn/wiki/Advanced-use-cases#file-icons),
- using the icons-in-terminal package (currently not packaged in homebrew)
- using Nerdfont fonts
- using Emoji

with the code [here](https://github.com/jarun/nnn/blob/1051d7213d910be4643d8736fea7a8b922916d3e/src/icons.h#L17-L24), the if-branch will ignore the emoji/nerd font options when the icons-in-terminal flag is set in the makefile, which is what happens in the current brew bottle. It would seem that this (having icons-in-terminal setup) is likely less likely for most average users to have the icons-in-terminal package, and it may be more likely that they'd have nerdfonts or emoji support. 

So may be change the build args to `O_NERD=1`?

**Comment by SMillerDev** [comment]
Could you make a pull request for this? It's really the best way
